pkg/virt-controller/watch/migration.go

```diff
@@ -668,6 +699,62 @@ func (c *MigrationController) handleTargetPodCreation(key string, migration *vir
 	return nil
 }
 
+func (c *MigrationController) createAttachmentPod(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, virtLauncherPod *k8sv1.Pod) error {
+	sourcePod, err := controller.CurrentVMIPod(vmi, c.podInformer)
+	if err != nil {
+		return fmt.Errorf("failed to get current VMI pod: %v", err)
+	}
+
+	volumes := getHotplugVolumes(vmi, sourcePod)
+	volumeNamesPVCMap := make(map[string]*k8sv1.PersistentVolumeClaim)
+	for _, volume := range volumes {
+		claimName := kubevirttypes.PVCNameFromVirtVolume(volume)
+		if claimName == "" {
+			log.Log.Object(migration).Infof("Unable to hotplug, volume %s not PVC or Datavolume", volume.Name)
+			continue
+		}
+		pvc, exists, _, err := kubevirttypes.IsPVCBlockFromStore(c.pvcInformer.GetStore(), virtLauncherPod.Namespace, claimName)
+		if err != nil {
+			return fmt.Errorf("failed to get pvc: %v", err)
+		}
+		if !exists {
+			log.Log.Object(migration).Infof("Unable to hotplug, claim %s not found", claimName)
+			continue
+		}
+		volumeNamesPVCMap[volume.Name] = pvc
+	}
```

Comment:
since this is duplicate logic in vmi.go, can we get a shared util function for this. I know it will require a bit refactoring.

Reply:
sure. will think how to refactor it.